### PR TITLE
Adding multi plot to make elemental analysis v 2.0

### LINKED ISSUE #24807: Elemental analysis and multiplot
Put the new multiplotting into elemental analysis

### PR DESCRIPTION
**Pull request by AnthonyLim23**
A new elemental analysis has been added and the old one has been renamed `Elemental_Analysis_Old`. This new version is designed to be easier to maintain and develop. It also uses the new multiplotting widget. 

The code will be refactored and unit tests added as part of #24926

**Report to:** [user name]/[nobody]. 

**To test:**
Contact me and I will send you some data (not to be distributed) 

1. Open Elemental Analysis
2. Click manage user directories and add the directory with the data to the search paths
3. In the run box enter 2695 and click load - data should be loaded
4. Clicking combinations of detectors, checkboxes and the periodic table will add lines to the plot. You will need to have a detector clicked to make the plot appear. 

Fixes #24807 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by NickDraper** [comment]
@AnthonyLim23 Could you take a look at the warning, and the conflict in the release notes

**Comment by AnthonyLim23** [comment]
the warning is not me, its in a cpp file, which I believe has been fixed.

**Comment by soininen** [comment]
The title of this PR should be changed to something more descriptive. At least the issue number need not be shown there.

File: scripts/MultiPlotting/multi_plotting_context.py [hunk 1]
```diff
@@ -30,20 +31,36 @@ def addSubplot(self, name, subplot):
...
+                self.subplots[subplotName].addLine(ws, specNum)
+
+            elif len(workspace) > 1:
                 self.subplots[subplotName].addLine(
                     workspace.OutputWorkspace, specNum)
             else:
                 self.subplots[subplotName].addLine(workspace, specNum)
         except:
-            print("cannot plot workspace")
+            print("cannot plot workspace", workspace)
```
Comment (soininen) [inline scripts/MultiPlotting/multi_plotting_context.py]: Shouldn't this error rather go to the logger? In any case, we should use Python 3 compatible printing: `print('cannot plot workspace {}'.format(workspace))`

File: scripts/MultiPlotting/multi_plotting_context.py [hunk 2]
```diff
@@ -30,20 +31,36 @@ def addSubplot(self, name, subplot):
...
         self.subplots[subplotName].add_annotate(label)
 
     def add_vline(self, subplotName, xvalue, name):
         self.subplots[subplotName].add_vline(xvalue, name)
 
+    def removeLabel(self, subplotName, name):
+        try:
+            self.subplots[subplotName].removeLabel(name)
+        except:
+            print("Could not remove label " + name)
```
Comment (soininen) [inline scripts/MultiPlotting/multi_plotting_context.py]: Similarly here: is this the optimal way to report errors?

**Review by soininen — CHANGES_REQUESTED** [review]
The plotting works well when it works. There seems to be a lot of error messages in Mantid's logs while playing around with the elemental analysis interface (non-Old). Here are a few:
- Checking GE1 and loading data
- Loading the proposed data and checking GE1 (or any other detector, it seems) for the first time
- Closing the plot window and unchecking the detectors

**Comment by AnthonyLim23** [comment]
To be honest most of those messages were put in to help me debug the code. I left them in as I thought it could be helpful. I could remove them

**Comment by soininen** [comment]
I see. I think they might scare the users so it may be a good idea to remove them, then.

**Comment by AnthonyLim23** [comment]
will do

**Review by soininen — APPROVED** [review]

**Comment by soininen** [comment]
Looks good to me. The failing IN5 detector efficiency system test is totally unrelated.

**Comment by NickDraper** [comment]
Created additional issue to improve release notes #25080

**Pull request merged**